neighbour-model/src/main/java/no/vegvesen/ixn/federation/model/Neighbour.java

```diff
@@ -235,7 +236,44 @@ public void setDnsProperties(Neighbour dnsNeighbour) {
 
 	}
 
-	public boolean shouldCheckSubscriptionRequestsForUpdates(LocalDateTime localSubscriptionsUpdated) {
+	/**
+	 * The basis of the subscription calculations are Self.local subscriptions and capabilities of the neighbour.
+	 * Should calculate new subscriptions if the localSubscriptionsUpdated time or the lastCapabilityExchange for
+	 * this neighbour is after the last time we calculated subscriptions and sent request to the neighbour.
+	 *
+	 * @param localSubscriptionsUpdated when are the last local subscriptions updated, optionally
+	 * @return true if the localSubscriptionsUpdated time or the getLastCapabilityExchange is after the last time we
+	 * calculated and sent subscription request to the neighbour
+	 */
+	public boolean shouldCheckSubscriptionRequestsForUpdates(Optional<LocalDateTime> localSubscriptionsUpdated) {
+		if (!localSubscriptionsUpdated.isPresent()) {
+			logger.debug("Local subscriptions never established for any service provider, should not check for subscription requests");
+			return false;
+		}
+		LocalDateTime localSubscriptionsUpdatedTime = localSubscriptionsUpdated.get();
+		logger.debug("Local subscription updated {}", localSubscriptionsUpdatedTime);
+
+		LocalDateTime maxUpdateDate;
+		if (getCapabilities() != null && this.getCapabilities().getLastCapabilityExchange() != null && this.getCapabilities().getLastCapabilityExchange().isAfter(localSubscriptionsUpdatedTime)) {
```

Comment:
Could this be made more explicit? Keep calling getCapabilities() and getCapabilties().getLastCapabiltyExchange over and over

Reply:
see my suggestion in commit a5c3c41